peer/addrinfo.go

```diff
@@ -61,6 +62,15 @@ func SplitAddr(m ma.Multiaddr) (transport ma.Multiaddr, id ID) {
 	return transport, id
 }
 
+func AddrInfoFromString(s string) (*AddrInfo, error) {
```

Comment:
You sure you don't want `AddrInfosFromStrings`?

Reply:
idk, this seems good enough for now? im happy with whatever